console_backend/src/ntrip_output.rs

```diff
@@ -0,0 +1,103 @@
+use crate::ntrip_tab::OutputType;
+use crate::types::{ArcBool, Result};
+use anyhow::Context;
+use crossbeam::channel::Receiver;
+use log::error;
+use std::io::Write;
+use std::process::{Command, Stdio};
+use std::{io, thread};
+
+pub struct MessageConverter {
+    in_rx: Receiver<Vec<u8>>,
+    running: ArcBool,
+    output_type: OutputType,
+}
+
+impl MessageConverter {
+    pub fn new(in_rx: Receiver<Vec<u8>>, output_type: OutputType) -> Self {
+        Self {
+            in_rx,
+            running: ArcBool::new(),
+            output_type,
+        }
+    }
+
+    pub fn start<W: Write + Send + 'static>(&mut self, out: W) -> Result<()> {
+        self.running.set(true);
+        match self.output_type {
+            OutputType::RTCM => self.output_rtcm(out),
+            OutputType::SBP => self.output_sbp(out),
+        }
+    }
+
+    /// Just redirects directly to writer
+    fn output_rtcm<W: Write + Send + 'static>(&mut self, mut out: W) -> Result<()> {
+        let in_rx = self.in_rx.clone();
+        let running = self.running.clone();
+        thread::spawn(move || loop {
+            if !running.get() {
+                break;
+            }
+            if let Ok(data) = in_rx.try_recv() {
+                if let Err(e) = out.write(&data) {
+                    error!("failed to write to device {e}");
+                }
+            }
+        });
+        Ok(())
+    }
+
+    /// Runs rtcm3tosbp converter
+    fn output_sbp<W: Write + Send + 'static>(&mut self, mut out: W) -> Result<()> {
+        let mut child = if cfg!(target_os = "windows") {
+            let mut cmd = Command::new("cmd");
+            cmd.args(["/C", "resources/binaries/win/rtcm3tosbp.exe"]);
+            cmd
+        } else if cfg!(target_os = "macos") {
+            let mut cmd = Command::new("sh");
+            cmd.args(["-c", "resources/binaries/mac/rtcm3tosbp"]);
+            cmd
+        } else {
+            let mut cmd = Command::new("sh");
+            cmd.args(["-c", "resources/binaries/win/rtcm3tosbp"]);
+            cmd
+        };
```

Comment:
MacOS is tested working.

(Also thinking alternative to embedding rtcm3tosbp by allowing selectable binary that consumes RTCM stdin and outputs stdout - then provide the binaries auto-download and just let users select path. If there are any changes to rtcm3tosbp or alternative format to send to device)